### ISSUE TITLE
Compression?

### ISSUE THREAD

**Issue by vinniefalco** [nayuki/Simple-DEFLATE-decompressor#1]
It looks like this implements **inflate** but where is the **deflate**? Where is the code to compress data?

**Comment by nayuki** [comment]
Yeah - sorry. DEFLATE is the name of the standard, but I only implemented the inflater (decompressor). My justification is that there is only one correct way to decompress the data (which has been implemented faithfully), but many possible ways to compress the data.

Choices for compressing the data range anywhere from using uncompressed blocks for all the data, to just doing Huffman coding on bytes (no LZ77), to doing some simple heuristical LZ77 dictionary matching, to very sophisticated "flexible parsing" LZ77 and making decisions about where to split DEFLATE blocks.
